pkg/profile/artifact.go

```diff
@@ -49,6 +56,84 @@ func NewProfilesArtifactsMaker(cfg MakerConfig) *ProfilesArtifactsMaker {
 	}
 }
 
+// GenerateArtifactsOutput takes a slice of generated artifacts and creates files, folders and cloned artifacts for them.
+func (pa *ProfilesArtifactsMaker) GenerateArtifactsOutput(artifacts []Artifact, profilesInstallations profilesv1.ProfileInstallation) error {
```

Comment:
Do we need `MakeArtifacts` and `GenerateArtifactsOutput` as two separate functions? They only ever get used in combination right? I know its easier for testing but it doesn't really serve much purpose. WDYT?

Reply:
Thought about that too, yes. I don't know. 😁 I kept it separate in the end because I thought about things like, we might want to keep the way we generate artifacts but change how we output them?